### PR TITLE
SpawnTheHero fix for neutral captives bug, plus enhancements

### LINKED ISSUE #80: BL Logic which releases SpawnHero Captives - Fix discussion
**Describe the bug**
Captive Wanderers released on generic Peace events.

**To Reproduce**
1. Spawn a wanderer prisoner
2. Peace with any faction
3. Wanderer is released

**Expected behavior**
Captive wanderers should not be at peace with Main Hero!!!

**Additional context**
>Hero hero = HeroCreator.CreateSpecialHero(wanderer, randomElementWithPredicate, **_CampaignData.NeutralFaction_**, CampaignData.NeutralFaction, -1);

This is the problem. This needs to pull in the Intended Clan. Let me see what I can find as far as correct syntax.

EDIT:
>Hero hero = HeroCreator.CreateSpecialHero(wanderer, randomElement, Clan.BanditFactions.GetRandomElementInefficiently(), null, -1);

You actually have this fixed in Master? Development reflects the first, erroneous pattern.

### PR DESCRIPTION
**Pull request by SkipWestcott**
Intent is to preserve StringId interpretation; Closes #80, begins to address #74

### CODE REVIEW

File: Events/CESpawnSystem.cs [hunk 1]
```diff
@@ -86,8 +86,27 @@ public void SpawnTheHero(SpawnHero[] variables, PartyBase party)
                     CharacterObject wanderer = cultureObject.NotableAndWandererTemplates.GetRandomElementWithPredicate((CharacterObject x) => x.Occupation == Occupation.Wanderer && (heroVariables.Gender == null || x.IsFemale == isFemale));
                     Settlement randomElement = Settlement.All.GetRandomElementWithPredicate((Settlement settlement) => settlement.Culture == wanderer.Culture && settlement.IsTown);
 
-                    Hero hero = HeroCreator.CreateSpecialHero(wanderer, randomElement, CampaignData.NeutralFaction, CampaignData.NeutralFaction, -1);
+                    Clan ceClan = Clan.BanditFactions.GetRandomElementInefficiently();
```
Comment (SkipWestcott) [inline Events/CESpawnSystem.cs:89]: Assume a random Bandit Clan.

File: Events/CESpawnSystem.cs [hunk 2]
```diff
@@ -86,8 +86,27 @@ public void SpawnTheHero(SpawnHero[] variables, PartyBase party)
...
+                        {
+                            case "captor":
+                                ceClan = party.Owner.Clan;
+                                break;
 
+                            case "player":
+                                ceClan = Clan.PlayerClan;
+                                break;
+
+                            default:
```
Comment (SkipWestcott) [inline Events/CESpawnSystem.cs:102]: Parse the variable["Clan"] for a Bandit Clan supplied.

**Comment by TheBadListener** [comment]
Is this ready to be merged?

**Comment by SkipWestcott** [comment]
> Is this ready to be merged? @SkipWestcott

100%, been running it locally all week!

**Pull request merged**